site/src/pages/HealthPage/DERPPage.tsx

```diff
@@ -0,0 +1,122 @@
+import { Link, useOutletContext } from "react-router-dom";
+import {
+  Header,
+  HeaderTitle,
+  Main,
+  SectionLabel,
+  BooleanPill,
+  Logs,
+  HealthyDot,
+} from "./Content";
+import { HealthcheckReport } from "api/typesGenerated";
+import Button from "@mui/material/Button";
+import LocationOnOutlined from "@mui/icons-material/LocationOnOutlined";
+import useTheme from "@mui/styles/useTheme";
+import { healthyColor } from "./healthyColor";
+import { Alert } from "components/Alert/Alert";
+import { Helmet } from "react-helmet-async";
+import { pageTitle } from "utils/page";
+
+const flags = [
+  "UDP",
+  "IPv6",
+  "IPv4",
+  "IPv6CanSend",
+  "IPv4CanSend",
+  "OSHasIPv6",
+  "ICMPv4",
+  "MappingVariesByDestIP",
+  "HairPinning",
+  "UPnP",
+  "PMP",
+  "PCP",
+];
+
+export const DERPPage = () => {
+  const { derp } = useOutletContext<HealthcheckReport>();
+  const { netcheck, regions, netcheck_logs: logs } = derp;
+  const theme = useTheme();
+
+  return (
+    <>
+      <Helmet>
+        <title>{pageTitle("DERP - Health")}</title>
+      </Helmet>
+
+      <Header>
+        <HeaderTitle>
+          <HealthyDot
+            healthy={derp.healthy}
+            hasWarnings={derp.warnings.length > 0}
+          />
+          DERP
+        </HeaderTitle>
+      </Header>
+
+      <Main>
+        {derp.warnings.map((warning, i) => {
+          return (
+            <Alert key={i} severity="warning">
+              {warning}
+            </Alert>
+          );
+        })}
+
+        <section>
+          <SectionLabel>Flags</SectionLabel>
+          <div css={{ display: "flex", flexWrap: "wrap", gap: 12 }}>
+            {flags.map((flag) => (
+              <BooleanPill key={flag} value={netcheck[flag]}>
+                {flag}
+              </BooleanPill>
+            ))}
+          </div>
+        </section>
+
+        <section>
+          <SectionLabel>Regions</SectionLabel>
+          <div css={{ display: "flex", flexWrap: "wrap", gap: 12 }}>
+            {Object.values(regions).map(({ region, healthy, warnings }) => {
+              return (
+                <Button
+                  startIcon={
+                    <LocationOnOutlined
+                      css={{
+                        width: 16,
+                        height: 16,
+                        color: healthyColor(
+                          theme,
+                          healthy,
+                          warnings?.length > 0,
```

Comment:
Learned something new – didn't realize this worked because if `warnings?.length` short-circuits to `undefined`, it gets coerced to `NaN`, and the math check still works

Reply:
We are now using severity instead of healthy so I change this completely